### ISSUE TITLE
off-by-one error in pndm scheduler

### ISSUE THREAD

**Issue by rokbok** [huggingface/diffusers#477]
There appears to be an off-by-one error on this line:
https://github.com/huggingface/diffusers/blob/25a51b63ca75e1351069bee87a0fb3df5abb89c3/src/diffusers/schedulers/scheduling_pndm.py#L338

I got `IndexError: index 1000 is out of bounds for dimension 0 with size 1000`

Repro script:
```import torch
from diffusers import StableDiffusionPipeline, StableDiffusionImg2ImgPipeline
from PIL import Image

pipe = StableDiffusionPipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4", 
    torch_dtype = torch.float16,
    revision = 'fp16',
    use_auth_token = True,
).to("cuda")

with torch.autocast("cuda"):
        image = pipe(prompt="astronaut or doesn't really matter", num_inference_steps=3)
```

**Comment by natolambert** [comment]
We are discussing this in #444 and is being solved in #466 soon!

**Comment by rokbok** [comment]
Thanks for the heads-up
